### PR TITLE
Feature/intercept dml without where

### PR DESCRIPTION
**Pull request by newborn22**
## Description
When the input SQL is converted into a stmt, it checks whether the WHERE clause is empty for  SQLs of `DELETE` and `UPDATE` types. 

This feature is **disabled by default**. Users can enable it by entering `set session enable_interception_for_dml_without_where='true'` and disable it by entering `set session enable_interception_for_dml_without_where='false'` **during the current session connection**. 

Users can also input set global `enable_interception_for_dml_without_where='true'; `or `set global enable_interception_for_dml_without_where='false';` to enable or disable this feature by default for **newly opened sessions**.

It will generates an error for DELETE and UPDATE SQLs with an empty WHERE clause.  

## Related Issue(s)
#334 

## Checklist
-  Tests were done by inputting DELETE and UPDATE sql without WHERE condition.

### CODE REVIEW

**Review by gerayking — COMMENTED** [review]
Should rebase main branch and re-run testcase.

File: go/vt/vtgate/plan_execute.go [hunk 1]
```diff
@@ -293,3 +300,26 @@ func (e *Executor) logPlanningFinished(logStats *logstats.LogStats, plan *engine
 	logStats.PlanTime = execStart.Sub(logStats.StartTime)
 	return execStart
 }
+
+func interceptDMLWithoutWhereEnable(safeSession *SafeSession, stmt sqlparser.Statement) error {
+	interceptDMLWithoutWhereEnable := false
+	enable, exist := safeSession.Session.UserDefinedVariables["interceptdmlwithoutwhere"]
```
Comment (earayu) [inline go/vt/vtgate/plan_execute.go:306]: param name here should not be case sensitive, 'intercept_dml_without_where' would be better

Comment (earayu) [reply go/vt/vtgate/plan_execute.go:306]: Should not use UserDefinedVariables here, see 'read_write_splitting_ratio' for reference

Comment (newborn22) [reply go/vt/vtgate/plan_execute.go]: the feature switch is defined as a system variable now and support `set session enable_interception_for_dml_without_where='true'` and `set global enable_interception_for_dml_without_where='true'`.

**Review by earayu — APPROVED** [review]
LGTM

**Pull request merged**